### PR TITLE
0016: Default worklow behaviour

### PR DESCRIPTION
**Pull request by thebsdbox**
## Description

This proposal defines the specifics for running a default workflow for unknown hardware

## Why is this needed

This is the beginning work for projects such as DCIM, or assisting in DataCentre OPS

### CODE REVIEW

**Comment by gianarb** [comment]
This is a lack of knowledge from my side but I always thought Tinkerbell had two different way of working.
The one I know and we have in the documentation requires each hardware to be registered. There is another way that should work differently, I assume like: "hardware not registered yet is capable of registering itself". If that's the case maybe some of this work is already done

**Comment by gianarb** [comment]
Since #21 we moved all the proposals inside the `proposals` directory. Do you mind to rebase and move your dir?
Thanks

**Comment by mmlb** [comment]
Tinkerbell does not have this built-in currently, but boots has something like what this proposal is ... proposing :D. Just not as automated, like at all.

We have recently landed the "Hardware Capture" part using https://github.com/packethost/discover-metal into OSIE in https://github.com/tinkerbell/osie/pull/150. Having tink default to a workflow that runs this image was part of the future plan already :D.

File: 0016/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+authors: Dan <daniel.finneran@gmail.com>
+---
+
+## Summary
+
+This additional feature would allow an end user to specify a workflow to be used against a machine that has no `hardware` profile specified within Tinkerbell. The main feature that this would enable would be for DCIM or hardware discovery where any new unknown machines are automatically prossed by a specified workflow that could perform actions such as:
+
+ - Secure wipe
+ - Forensic analysis
+ - DataCentre inventory
```
Comment (mmlb) [inline 0016/README.md]: What kind of DC inventory are you thinking? Presumably an untrusted machine shouldn't have network access to all parts of a DC.

**Comment by gianarb** [comment]
Ok now that we know this feature is not there I think we should merge and accept this issue. @thebsdbox do you mind to do a  review of the questions, update status to accepted and move to the right directory?

**Review by gianarb — CHANGES_REQUESTED** [review]
Ok now that we know this feature is not there I think we should merge and accept this issue. @thebsdbox do you mind to do a review of the questions, update status to accepted and move to the right directory?

File: 0016/README.md (hunk 1 above)
Comment (thebsdbox) [reply 0016/README.md]: The Datacentre inventory would just be whatever tinker bell is capturing from the machines that boot the first time when we don't know about them.

- NICs
- CPU
- Memory 
- GPU
- Disk config
- Is anything installed on the servers
- Firmwares?

Comment (mmlb) [reply 0016/README.md]: ah well that doesn't sound like DC inventory, more like node inventory right? Just a wording that might be confusing.

**Review by mmlb — DISMISSED** [review]

**Review by gianarb — DISMISSED** [review]

**Review by gianarb — APPROVED** [review]

**Pull request merged**